diff --git a/Makefile b/Makefile
index 40a6387..c37812b 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ TARGET = mini_telnetd
 
 all: $(TARGET)
 
-CFLAGS = -O2 -Wall
+CFLAGS = -g -O0 -Wall
 # Static compile
 #CFLAGS = -O2 -Wall -static -static-libgcc
 # Use for older systems 
diff --git a/mini_telnetd.c b/mini_telnetd.c
index bd29053..0fc15a0 100644
--- a/mini_telnetd.c
+++ b/mini_telnetd.c
@@ -20,6 +20,12 @@
  * Vladimir Oleynik <dzo@simtreas.ru> 2001
  *     Set process group corrections, initial busybox port
  *
+ * Nick Glazzard <nick.nos28@gmail.com> 2025
+ *     Fix errexit() to use varargs so it actually works.
+ *     Add sys/stat.h so it compiles.
+ *     Change XTABS to TAB3 in termios options so it compiles.
+ *     Now works on macOS 15.5.
+ *
  * BusyBox is distributed under version 2 of the General Public License please 
  * see the file LICENSE for further information. Version 2 is the only version
  * of this license which this version of BusyBox
@@ -45,6 +51,9 @@
 #include <ctype.h>
 #include <sys/syslog.h>
 
+#include <sys/stat.h>
+#include <stdarg.h>
+
 #define BUFSIZE 4000
 #define SOCKET_TYPE	AF_INET
 #define MIN(X,Y) ((X) < (Y) ? (X) : (Y)) 
@@ -221,6 +230,7 @@ static int getpty(char *line)
                         }
                 }
         }
+        return -1;
 #endif
 }
 
@@ -235,12 +245,13 @@ static void send_iac(struct tsession *ts, unsigned char command, int option)
 	ts->size2 += 3;
 }
 
-
-
-void errexit(const char *s, ...)
+void errexit(const char *fmt, ...)
 {
-	fprintf( stderr, s );
-	exit( EXIT_FAILURE );
+  va_list args;
+  va_start(args, fmt);
+  vfprintf(stderr, fmt, args);
+  va_end(args);
+  exit( EXIT_FAILURE );
 }
 
 static struct tsession *make_new_session(int sockfd)
@@ -311,7 +322,8 @@ static struct tsession *make_new_session(int sockfd)
 
 		tcgetattr(0, &termbuf);
 		termbuf.c_lflag |= ECHO; /* if we use readline we dont want this */
-		termbuf.c_oflag |= ONLCR|XTABS;
+		/*termbuf.c_oflag |= ONLCR|XTABS;*/
+		termbuf.c_oflag |= ONLCR|TAB3;
 		termbuf.c_iflag |= ICRNL;
 		termbuf.c_iflag &= ~IXOFF;
 		/*termbuf.c_lflag &= ~ICANON;*/
